components/style/properties/helpers.mako.rs

```diff
@@ -122,6 +124,14 @@
             % if need_animatable or animation_value_type == "ComputedValue":
             #[derive(Animate, ComputeSquaredDistance)]
             % endif
+            % if not allow_empty:
```

Comment:
I feel ambivalent about this one... As long as we need to keep the `allow_empty` one this doesn't seem particularly simpler in any way.